src/plugins/totle.js

```diff
@@ -0,0 +1,225 @@
+        toTokenAddress = toToken.address
+      } else if (request.toCurrencyCode === 'ETH') {
+        fromTokenAddress = fromToken.address
+        toTokenAddress = '0x0000000000000000000000000000000000000000'
+      } else if (!fromToken || !toToken) {
+        throw new SwapCurrencyError(
+          swapInfo,
+          request.fromCurrencyCode,
+          request.toCurrencyCode
+        )
+      } else {
+        fromTokenAddress = fromToken.address
+        toTokenAddress = toToken.address
+      }
+
+      // Grab addresses:
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+
+      // Swap the currencies if we need a reverse quote:
+      const quoteParams =
+        request.quoteFor === 'from'
+          ? {
+            from: fromTokenAddress,
+            to: toTokenAddress
+          }
+          : {
+            from: toTokenAddress,
+            to: fromTokenAddress
+          }
+
+      // Get the estimate from the server:
+      const reply = await call({
+        address: fromAddress,
+        affiliateContract,
+        swap: {
+          ...quoteParams,
+          amount: request.nativeAmount,
+          minFillPercent: 97,
+          minSlippagePercent: 2
+        }
+      })
+      // TODO: check
+      checkReply(reply)
+
+      const quoteInfo: QuoteInfo = reply.response
+
+      let fromNativeAmount = request.nativeAmount
+      let toNativeAmount = '0'
+      if (quoteInfo.summary.sells) {
+        let fromAmount = '0'
+        for (const { amount } of quoteInfo.summary.sells) {
+          fromAmount = (Number(fromAmount) + Number(amount)).toString()
+        }
+        fromNativeAmount = fromAmount
+      }
+      if (quoteInfo.summary.buys) {
+        for (const { amount } of quoteInfo.summary.buys) {
+          toNativeAmount = (Number(toNativeAmount) + Number(amount)).toString()
+        }
+      }
+
+      // Make the transaction:
+      const spendInfo = {
+        currencyCode: request.fromCurrencyCode,
+        spendTargets: [
+          {
+            nativeAmount: quoteInfo.ethValue,
+            publicAddress: quoteInfo.contractAddress,
+            otherParams: {
+              pluginName: swapInfo.pluginName,
+              uniqueIdentifier: quoteInfo.id,
+              data: quoteInfo.payload.data
+            }
+          }
+        ],
+        networkFeeOption: 'custom',
+        customNetworkFee: {
+          gasLimit: quoteInfo.gas.limit,
+          gasPrice: quoteInfo.gas.price
+        }
+      }
+      io.console.info('Totle spendInfo', spendInfo)
+      const tx = await request.fromWallet.makeSpend(spendInfo)
+      tx.otherParams.payinAddress = spendInfo.spendTargets[0].publicAddress
+      tx.otherParams.uniqueIdentifier =
+        spendInfo.spendTargets[0].otherParams.uniqueIdentifier
+
+      return makeSwapPluginQuote(
```

Comment:
Since this plugin needs to send not one, but two transactions, I would just extract the contents of `makeSwapPluginQuote` and do that work inline here.

Reply:
👍 I actually already made this change and testing it out rn. Will hopefully push this change today.